src/sparse/impl/KokkosSparse_spmv_impl.hpp

```diff
@@ -567,315 +1116,83 @@ struct SPMV_MV_LayoutLeft_Functor {
 #pragma unroll
 #endif
 #ifdef KOKKOS_ENABLE_PRAGMA_LOOPCOUNT
-#pragma loop count (15)
+#pragma loop count(15)
 #endif
-#ifdef __CUDA_ARCH__
-        for (ordinal_type iEntry = static_cast<ordinal_type> (threadIdx.x);
-             iEntry < row.length;
-             iEntry += static_cast<ordinal_type> (blockDim.x))
-#else
-        for (ordinal_type iEntry = 0;
-             iEntry < row.length;
-             iEntry ++)
-#endif
-      {
-      const A_value_type val = conjugate ?
-        Kokkos::Details::ArithTraits<A_value_type>::conj (row.value(iEntry)) :
-        row.value(iEntry);
-      const ordinal_type ind = row.colidx(iEntry);
+    for(int entryIdx = 0; entryIdx < row.length; ++entryIdx) {
```

Comment:
The right loop index type here should be `ordinal_type`, since it is large enough to hold all the entries in a row (assuming no duplicates).